editor/filesystem_dock.cpp

```diff
@@ -1812,11 +1812,8 @@ void FileSystemDock::_file_option(int p_option, const Vector<String> &p_selected
 				fpath = p_selected[0];
 			}
 
-			if (!fpath.ends_with("/")) {
-				fpath = fpath.get_base_dir();
-			}
 			String dir = ProjectSettings::get_singleton()->globalize_path(fpath);
-			OS::get_singleton()->shell_open(String("file://") + dir);
+			OS::get_singleton()->shell_show_in_explorer(dir);
```

Comment:
There are much more code like this (`OS::get_singleton()->shell_open(String("file://") + ...)`) within the codebase (e.g. in the `editor_node.cpp`), likely all these should be changed to use the new method.

Reply:
OK, I replaced them, and add a param `open_folder` to decide whether open the folder instead of navigate to it's top level folder and select it( only when the provided path is a valid directory path).
Most behavior of existed  `OS::get_singleton()->shell_open(String("file://") + ...)` will keep old behavior.